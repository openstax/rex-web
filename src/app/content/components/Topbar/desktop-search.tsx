import React from 'react';
import * as Styled from './styled';
import { mobileNudgeStudyToolsTargetId } from '../NudgeStudyTools/constants';
import { NudgeElementTarget } from '../NudgeStudyTools/styles';
import type { SearchArgs } from './search-common';

type Args = React.PropsWithChildren<
  SearchArgs & { openMenu(e: React.MouseEvent<HTMLButtonElement>): void }
>;

export default function DesktopSearch({
  state,
  onSearchSubmit,
  onSearchClear,
  onSearchChange,
  colorSchema,
  searchInSidebar,
  mobileToolbarOpen,
  newButtonEnabled,
  toggleMobile,
  openMenu,
  children,
}: Args) {
  return (
    <Styled.SearchPrintWrapper>
      <NudgeElementTarget id={mobileNudgeStudyToolsTargetId}>
        <Styled.MenuButton type='button' onClick={openMenu} />
      </NudgeElementTarget>

      <Styled.SearchInputWrapper
        active={mobileToolbarOpen}
        onSubmit={onSearchSubmit}
        data-testid='desktop-search'
        data-experiment
        colorSchema={colorSchema}
        searchInSidebar={searchInSidebar}
      >
        <Styled.SearchInput
          desktop
          type='search'
          data-testid='desktop-search-input'
          onChange={onSearchChange}
          value={state.query}
        />
        <Styled.SearchButton
          mobile
          type='button'
          ariaLabelId='i18n:toolbar:search:toggle'
          data-analytics-label='Search this book'
          data-testid='mobile-toggle'
          data-experiment
          onClick={toggleMobile}
          colorSchema={colorSchema}
        />
        {!state.formSubmitted && !newButtonEnabled && (
          <Styled.SearchButton
            desktop
            colorSchema={colorSchema}
            data-experiment
          />
        )}
        {state.formSubmitted && !newButtonEnabled && (
          <Styled.CloseButton
            desktop
            type='button'
            onClick={onSearchClear}
            data-testid='desktop-clear-search'
          />
        )}
        {state.formSubmitted && newButtonEnabled && (
          <Styled.CloseButtonNew
            desktop
            type='button'
            onClick={onSearchClear}
            data-testid='desktop-clear-search'
          >
            <Styled.CloseIcon />
          </Styled.CloseButtonNew>
        )}
        {newButtonEnabled && (
          <Styled.SearchButton
            desktop
            colorSchema={colorSchema}
            data-experiment
          />
        )}
      </Styled.SearchInputWrapper>
      {children}
    </Styled.SearchPrintWrapper>
  );
}
