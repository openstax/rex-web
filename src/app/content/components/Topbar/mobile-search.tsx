import React from 'react';
import * as Styled from './styled';
import { FormattedMessage } from 'react-intl';
import type { SearchArgs } from './search-common';

type Args = React.PropsWithChildren<
  SearchArgs & { showBackToSearchResults: boolean }
>;

export default function MobileSearch({
  showBackToSearchResults,
  onSearchSubmit,
  onSearchClear,
  onSearchChange,
  colorSchema,
  searchInSidebar,
  mobileToolbarOpen,
  openSearchResults,
  toggleMobile,
  state,
  newButtonEnabled,
  children,
}: Args) {
  const openSearchbar = React.useCallback(
    (e: React.MouseEvent<HTMLButtonElement>) => {
      e.preventDefault();
      openSearchResults();
    },
    [openSearchResults]
  );

  return (
    <Styled.MobileSearchWrapper mobileToolbarOpen={mobileToolbarOpen}>
      <Styled.Hr />
      <Styled.MobileSearchContainer>
        {showBackToSearchResults && (
          <FormattedMessage id='i18n:search-results:bar:toggle-text:mobile'>
            {(msg) => (
              <Styled.SeachResultsTextButton
                onClick={openSearchbar}
                data-testid='back-to-search-results'
              >
                <Styled.LeftArrow />
                <Styled.InnerText>{msg}</Styled.InnerText>
              </Styled.SeachResultsTextButton>
            )}
          </FormattedMessage>
        )}
        {!showBackToSearchResults && (
          <FormattedMessage id='i18n:search-results:bar:close-text:mobile'>
            {(msg) => (
              <Styled.CloseSearchResultsTextButton
                onClick={toggleMobile}
                data-testid='close-search-results'
              >
                <Styled.InnerText>{msg}</Styled.InnerText>
              </Styled.CloseSearchResultsTextButton>
            )}
          </FormattedMessage>
        )}
        <Styled.SearchInputWrapper
          action='#'
          onSubmit={onSearchSubmit}
          data-testid='mobile-search'
          data-experiment
          colorSchema={colorSchema}
          searchInSidebar={searchInSidebar}
        >
          <Styled.SearchInput
            mobile
            type='search'
            data-testid='mobile-search-input'
            autoFocus
            onChange={onSearchChange}
            value={state.query}
          />
          {!state.formSubmitted && !newButtonEnabled && (
            <Styled.SearchButton
              desktop
              colorSchema={colorSchema}
              data-experiment
            />
          )}
          {state.query && newButtonEnabled && (
            <Styled.CloseButtonNew
              type='button'
              onClick={onSearchClear}
              formSubmitted={state.formSubmitted}
              data-testid='mobile-clear-search'
            >
              <Styled.CloseIcon />
            </Styled.CloseButtonNew>
          )}
          {state.query && !newButtonEnabled && (
            <Styled.CloseButton
              type='button'
              onClick={onSearchClear}
              formSubmitted={state.formSubmitted}
              data-testid='mobile-clear-search'
            />
          )}
        </Styled.SearchInputWrapper>
        {children}
      </Styled.MobileSearchContainer>
    </Styled.MobileSearchWrapper>
  );
}
