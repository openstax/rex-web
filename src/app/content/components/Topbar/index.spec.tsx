import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Topbar from '.';
import createTestServices from '../../../../test/createTestServices';
import createTestStore from '../../../../test/createTestStore';
import MessageProvider from '../../../../test/MessageProvider';
import { book as archiveBook } from '../../../../test/mocks/archiveLoader';
import { mockCmsBook } from '../../../../test/mocks/osWebLoader';
import { makeEvent, makeFindByTestId, makeFindOrNullByTestId, makeInputEvent } from '../../../../test/reactutils';
import { makeSearchResults } from '../../../../test/searchResults';
import TestContainer from '../../../../test/TestContainer';
import * as Services from '../../../context/Services';
import { MiddlewareAPI, Store } from '../../../types';
import { assertDocument } from '../../../utils';
import { openMobileMenu } from '../../actions';
import {
  clearSearch,
  closeSearchResultsMobile,
  openSearchResultsMobile,
  receiveSearchResults,
  requestSearch
} from '../../search/actions';
import * as searchSelectors from '../../search/selectors';
import { formatBookData } from '../../utils';
import { CloseButtonNew, MenuButton, SearchButton } from './styled';

const book = formatBookData(archiveBook, mockCmsBook);

describe('search', () => {
  let store: Store;
  let dispatch: jest.SpyInstance;
  let services: ReturnType<typeof createTestServices> & MiddlewareAPI;

  beforeEach(() => {
    store = createTestStore();
    dispatch = jest.spyOn(store, 'dispatch');
    services = {
      ...createTestServices(),
      dispatch: store.dispatch,
      getState: store.getState,
    };
  });

  const render = () => renderer.create(<Provider store={store}>
    <Services.Provider value={services}>
      <MessageProvider>
        <Topbar />
      </MessageProvider>
    </Services.Provider>
  </Provider>);

  it('opens and closes mobile interface', () => {
    const component = render();
    const findById = makeFindByTestId(component.root);

    const event = makeEvent();

    renderer.act(() => {
      findById('mobile-toggle').props.onClick(event);
    });
    expect(() => findById('mobile-search')).not.toThrow();
    renderer.act(() => {
      findById('mobile-toggle').props.onClick(event);
    });
    expect(() => findById('mobile-search')).toThrow();
    expect(event.preventDefault).toHaveBeenCalledTimes(2);
  });

  it('doesn\'t dispatch search for empty string', () => {
    const component = render();
    const findById = makeFindByTestId(component.root);

    const inputEvent = makeInputEvent('');
    renderer.act(() => {
      findById('desktop-search-input').props.onChange(inputEvent);
    });

    const event = makeEvent();
    renderer.act(() => {
      findById('desktop-search').props.onSubmit(event);
    });
    expect(event.preventDefault).toHaveBeenCalled();

    expect(dispatch).not.toHaveBeenCalledWith(requestSearch('cool search'));
  });

  it('loads search query into text input', () => {
    const component = render();
    const findById = makeFindByTestId(component.root);

    expect(findById('desktop-search-input').props.value).toEqual('');
    renderer.act(() => {
      store.dispatch(requestSearch('cool search'));
    });
    expect(findById('desktop-search-input').props.value).toEqual('cool search');
  });

  it('tries to blur the focused element on submit', () => {
    const component = render();
    const document = assertDocument();
    const findById = makeFindByTestId(component.root);

    const inputEvent = makeInputEvent('cool search');
    renderer.act(() => {
      findById('desktop-search-input').props.onChange(inputEvent);
    });

    const htmlelement = document.createElement('div');
    Object.defineProperty(document, 'activeElement', {value: htmlelement, writable: true});
    const blur1 = jest.spyOn(htmlelement, 'blur');
    renderer.act(() => findById('desktop-search').props.onSubmit(makeEvent()));
    expect(blur1).toHaveBeenCalled();

    // test non HTMLElement branch
    const svgelement = document.createElementNS('http://www.w3.org/2000/svg', 'path');
    Object.defineProperty(document, 'activeElement', {value: svgelement});
    const blur2 = jest.spyOn(svgelement, 'blur');
    renderer.act(() => findById('desktop-search').props.onSubmit(makeEvent()));
    expect(blur2).not.toHaveBeenCalled();
  });

  it('search and clear work on desktop', () => {
    const component = render();
    const findById = makeFindByTestId(component.root);

    const inputEvent = makeInputEvent('cool search');
    findById('desktop-search-input').props.onChange(inputEvent);

    const event = makeEvent();
    renderer.act(() => findById('desktop-search').props.onSubmit(event));
    expect(event.preventDefault).toHaveBeenCalled();

    renderer.act(() => {
      expect(dispatch).toHaveBeenCalledWith(requestSearch('cool search'));
    });

    expect(findById('desktop-search-input').props.value).toEqual('cool search');

    const clearClick = makeEvent();
    renderer.act(() => {
      findById('desktop-clear-search').props.onClick(clearClick);
    });
    expect(clearClick.preventDefault).toHaveBeenCalled();

    expect(findById('desktop-search-input').props.value).toEqual('');
  });

  it('search and clear work on mobile', () => {
    const component = render();
    const findById = makeFindByTestId(component.root);

    renderer.act(() => {
      findById('mobile-toggle').props.onClick(makeEvent());
    });

    const inputEvent = makeInputEvent('cool search');
    renderer.act(() => {
      findById('mobile-search-input').props.onChange(inputEvent);
    });

    const event = makeEvent();
    renderer.act(() => findById('mobile-search').props.onSubmit(event));
    expect(event.preventDefault).toHaveBeenCalled();

    renderer.act(() => {
      expect(dispatch).toHaveBeenCalledWith(requestSearch('cool search'));
    });

    expect(findById('mobile-search-input').props.value).toEqual('cool search');

    const clearClick = makeEvent();
    renderer.act(() => {
      findById('mobile-clear-search').props.onClick(clearClick);
    });
    expect(clearClick.preventDefault).toHaveBeenCalled();

    expect(findById('mobile-search-input').props.value).toEqual('');
  });

  it('shows the "back to results" link after selecting a result', () => {
    const component = render();
    const findById = makeFindByTestId(component.root);
    const findOrNull = makeFindOrNullByTestId(component.root);

    expect(findOrNull('back-to-search-results')).toBe(null);

    renderer.act(() => {
      store.dispatch(requestSearch('cool search'));
      store.dispatch(receiveSearchResults(makeSearchResults()));
      store.dispatch(closeSearchResultsMobile());
    });

    expect(findById('back-to-search-results')).toBeTruthy();
  });

  it('clicking "back to results" opens results', () => {
    const component = render();
    const findById = makeFindByTestId(component.root);

    renderer.act(() => {
      store.dispatch(requestSearch('cool search'));
      store.dispatch(receiveSearchResults(makeSearchResults()));
      store.dispatch(closeSearchResultsMobile());
    });

    expect(dispatch).not.toHaveBeenCalledWith(openSearchResultsMobile());

    renderer.act(() => {
      findById('back-to-search-results').props.onClick(makeEvent());
    });

    expect(dispatch).toHaveBeenCalledWith(openSearchResultsMobile());
  });

  it('clicking "Close" clears search results', () => {
    const component = render();
    const findById = makeFindByTestId(component.root);

    renderer.act(() => {
      store.dispatch(requestSearch('cool search'));
      store.dispatch(receiveSearchResults(makeSearchResults()));
    });

    renderer.act(() => {
      findById('close-search-results').props.onClick(makeEvent());
    });

    expect(dispatch).toHaveBeenCalledWith(clearSearch());
  });

  it('input value syncs between mobile and desktop', () => {
    const component = render();
    const findById = makeFindByTestId(component.root);

    const desktopInputEvent = makeInputEvent('cool search');
    renderer.act(() => {
      findById('desktop-search-input').props.onChange(desktopInputEvent);

      findById('mobile-toggle').props.onClick(makeEvent());
    });

    expect(findById('mobile-search-input').props.value).toEqual('cool search');
    const inputEvent = makeInputEvent('asdf');
    renderer.act(() => {
      findById('mobile-search-input').props.onChange(inputEvent);

      findById('mobile-toggle').props.onClick(makeEvent());
    });

    expect(findById('desktop-search-input').props.value).toEqual('asdf');
  });
});

describe('search button', () => {
  let store: Store;

  beforeEach(() => {
    store = createTestStore();
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  const render = () => renderer.create(<TestContainer store={store}><Topbar /></TestContainer>);

  it('button has theme bg color applied', () => {
    const color = searchSelectors.searchButtonColor.resultFunc('bannerColorButton', book, 'blue');
    jest.spyOn(searchSelectors, 'searchButtonColor').mockReturnValue(color);
    jest.spyOn(searchSelectors, 'mobileToolbarOpen').mockReturnValue(true);

    const component = render();
    const [searchButton, searchButtonMobile] = component.root.findAllByType(SearchButton);

    expect(searchButton.props.colorSchema).toEqual('blue');
    expect(searchButtonMobile.props.colorSchema).toEqual('blue');
  });

  it('button has gray bg color applied', () => {
    const color = searchSelectors.searchButtonColor.resultFunc('grayButton', book, 'red');
    jest.spyOn(searchSelectors, 'searchButtonColor').mockReturnValue(color);

    const component = render();
    const [searchButton, searchButtonMobile] = component.root.findAllByType(SearchButton);

    expect(searchButton.props.colorSchema).toEqual('gray');
    expect(searchButtonMobile.props.colorSchema).toEqual('gray');
  });

  it('button has no bg color applied', () => {
    const color = searchSelectors.searchButtonColor.resultFunc(null, book, 'blue');
    jest.spyOn(searchSelectors, 'searchButtonColor').mockReturnValue(color);

    const component = render();
    const [searchButton, searchButtonMobile] = component.root.findAllByType(SearchButton);

    expect(searchButton.props.colorSchema).toEqual(null);
    expect(searchButtonMobile.props.colorSchema).toEqual(null);
  });

  it('new search buttons use new close button', () => {
    jest.spyOn(searchSelectors, 'searchButtonColor').mockReturnValue('green');

    const component = render();
    const findById = makeFindByTestId(component.root);

    const inputEvent = makeInputEvent('cool search');
    findById('desktop-search-input').props.onChange(inputEvent);

    const event = makeEvent();
    renderer.act(() => findById('desktop-search').props.onSubmit(event));

    const [closeButton] = component.root.findAllByType(CloseButtonNew);

    expect(closeButton).toBeTruthy();
  });
});

describe('mobile menu button', () => {
  let store: Store;
  let dispatch: jest.SpyInstance;

  beforeEach(() => {
    store = createTestStore();
    dispatch = jest.spyOn(store, 'dispatch');
  });

  it('opens mobile menu', () => {
    const component = renderer.create(<TestContainer store={store}><Topbar /></TestContainer>);

    renderer.act(() => {
      component.root.findByType(MenuButton).props.onClick({preventDefault: jest.fn()});
    });

    expect(dispatch).toHaveBeenCalledWith(openMobileMenu());
  });
});
