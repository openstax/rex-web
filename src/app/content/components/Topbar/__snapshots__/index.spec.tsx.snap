// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text resizer does not render if textSize is null 1`] = `
<div
  className="styled__TopBarWrapper-sc-12dq39v-0 aQNQa"
  data-testid="topbar"
>
  <div
    className="styled__SearchPrintWrapper-sc-12dq39v-9 dHOmyE"
  >
    <div
      className="styles__NudgeElementTarget-hrv0cf-10 BGmTi"
      id="mobile-menu-button"
    >
      <button
        aria-label="Click to open menu"
        className="styled__MenuButton-sc-12dq39v-2 bDNWEV styled__PlainButton-sc-1119auh-0 kPdLbJ"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="styled__HamburgerIcon-sc-12dq39v-1 fKwlfW StyledIconBase-sc-bdy9j4 cCvWXa"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 448 512"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <form
      className="styled__SearchInputWrapper-sc-12dq39v-7 bHZhxZ"
      data-experiment={true}
      data-testid="desktop-search"
      onSubmit={[Function]}
    >
      <input
        aria-label="Search this book"
        className="styled__SearchInput-sc-12dq39v-8 kGWMSU"
        data-testid="desktop-search-input"
        onChange={[Function]}
        placeholder="Search this book"
        type="search"
        value=""
      />
      <button
        aria-label="Search this book"
        className="styled__SearchButton-sc-12dq39v-3 dAFIlk styled__PlainButton-sc-1119auh-0 kPdLbJ"
        data-analytics-label="Search this book"
        data-experiment={true}
        data-testid="mobile-toggle"
        onClick={[Function]}
        type="button"
        value="Search"
      >
        <svg
          aria-hidden="true"
          height="18px"
          version="1.1"
          viewBox="0 0 18 18"
          width="18px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              fillRule="nonzero"
              transform="translate(-938.000000, -206.000000)"
            >
              <path
                d="M945,206 C948.865993,206 952,209.134007 952,213 C952,214.662837 951.420201,216.190255 950.45174,217.391118 L955.03033,221.96967 C955.323223,222.262563 955.323223,222.737437 955.03033,223.03033 C954.764064,223.296597 954.3474,223.320803 954.053788,223.102948 L953.96967,223.03033 L949.391118,218.45174 C948.190255,219.420201 946.662837,220 945,220 C941.134007,220 938,216.865993 938,213 C938,209.134007 941.134007,206 945,206 Z M945,207.5 C941.962434,207.5 939.5,209.962434 939.5,213 C939.5,216.037566 941.962434,218.5 945,218.5 C948.037566,218.5 950.5,216.037566 950.5,213 C950.5,209.962434 948.037566,207.5 945,207.5 Z"
              />
            </g>
          </g>
        </svg>
      </button>
      <button
        className="styled__SearchButton-sc-12dq39v-3 eoLjh styled__PlainButton-sc-1119auh-0 kPdLbJ"
        data-experiment={true}
        value="Search"
      >
        <svg
          aria-hidden="true"
          height="18px"
          version="1.1"
          viewBox="0 0 18 18"
          width="18px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              fillRule="nonzero"
              transform="translate(-938.000000, -206.000000)"
            >
              <path
                d="M945,206 C948.865993,206 952,209.134007 952,213 C952,214.662837 951.420201,216.190255 950.45174,217.391118 L955.03033,221.96967 C955.323223,222.262563 955.323223,222.737437 955.03033,223.03033 C954.764064,223.296597 954.3474,223.320803 954.053788,223.102948 L953.96967,223.03033 L949.391118,218.45174 C948.190255,219.420201 946.662837,220 945,220 C941.134007,220 938,216.865993 938,213 C938,209.134007 941.134007,206 945,206 Z M945,207.5 C941.962434,207.5 939.5,209.962434 939.5,213 C939.5,216.037566 941.962434,218.5 945,218.5 C948.037566,218.5 950.5,216.037566 950.5,213 C950.5,209.962434 948.037566,207.5 945,207.5 Z"
              />
            </g>
          </g>
        </svg>
      </button>
    </form>
  </div>
  <div
    className="styled__MobileSearchWrapper-sc-12dq39v-11 foGdGm"
  >
    <hr
      className="styled__Hr-sc-12dq39v-12 iJYKyP"
    />
    <div
      className="styled__MobileSearchContainer-sc-12dq39v-10 hKhfgu"
    >
      <button
        className="styled__PlainButton-sc-1119auh-0 styled__SeachResultsTextButton-sc-12dq39v-15 styled__CloseSearchResultsTextButton-sc-12dq39v-19 kzdVbH"
        data-testid="close-search-results"
        onClick={[Function]}
      >
        <div
          className="styled__InnerText-sc-12dq39v-14 fahaRL"
        >
          Close
        </div>
      </button>
      <form
        action="#"
        className="styled__SearchInputWrapper-sc-12dq39v-7 bHZhxZ"
        data-experiment={true}
        data-testid="mobile-search"
        onSubmit={[Function]}
      >
        <input
          aria-label="Search this book"
          autoFocus={true}
          className="styled__SearchInput-sc-12dq39v-8 bHFdA"
          data-testid="mobile-search-input"
          onChange={[Function]}
          placeholder="Search this book"
          type="search"
          value=""
        />
        <button
          className="styled__SearchButton-sc-12dq39v-3 eoLjh styled__PlainButton-sc-1119auh-0 kPdLbJ"
          data-experiment={true}
          value="Search"
        >
          <svg
            aria-hidden="true"
            height="18px"
            version="1.1"
            viewBox="0 0 18 18"
            width="18px"
          >
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <g
                fill="currentColor"
                fillRule="nonzero"
                transform="translate(-938.000000, -206.000000)"
              >
                <path
                  d="M945,206 C948.865993,206 952,209.134007 952,213 C952,214.662837 951.420201,216.190255 950.45174,217.391118 L955.03033,221.96967 C955.323223,222.262563 955.323223,222.737437 955.03033,223.03033 C954.764064,223.296597 954.3474,223.320803 954.053788,223.102948 L953.96967,223.03033 L949.391118,218.45174 C948.190255,219.420201 946.662837,220 945,220 C941.134007,220 938,216.865993 938,213 C938,209.134007 941.134007,206 945,206 Z M945,207.5 C941.962434,207.5 939.5,209.962434 939.5,213 C939.5,216.037566 941.962434,218.5 945,218.5 C948.037566,218.5 950.5,216.037566 950.5,213 C950.5,209.962434 948.037566,207.5 945,207.5 Z"
                />
              </g>
            </g>
          </svg>
        </button>
        
        
      </form>
    </div>
  </div>
</div>
`;

exports[`text resizer opens menu when clicking menu button 1`] = `
<div
  className="styled__TopBarWrapper-sc-12dq39v-0 aQNQa"
  data-testid="topbar"
>
  <div
    className="styled__SearchPrintWrapper-sc-12dq39v-9 dHOmyE"
  >
    <div
      className="styles__NudgeElementTarget-hrv0cf-10 BGmTi"
      id="mobile-menu-button"
    >
      <button
        aria-label="Click to open menu"
        className="styled__MenuButton-sc-12dq39v-2 bDNWEV styled__PlainButton-sc-1119auh-0 kPdLbJ"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="styled__HamburgerIcon-sc-12dq39v-1 fKwlfW StyledIconBase-sc-bdy9j4 cCvWXa"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 448 512"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <form
      className="styled__SearchInputWrapper-sc-12dq39v-7 bHZhxZ"
      data-experiment={true}
      data-testid="desktop-search"
      onSubmit={[Function]}
    >
      <input
        aria-label="Search this book"
        className="styled__SearchInput-sc-12dq39v-8 kGWMSU"
        data-testid="desktop-search-input"
        onChange={[Function]}
        placeholder="Search this book"
        type="search"
        value=""
      />
      <button
        aria-label="Search this book"
        className="styled__SearchButton-sc-12dq39v-3 dAFIlk styled__PlainButton-sc-1119auh-0 kPdLbJ"
        data-analytics-label="Search this book"
        data-experiment={true}
        data-testid="mobile-toggle"
        onClick={[Function]}
        type="button"
        value="Search"
      >
        <svg
          aria-hidden="true"
          height="18px"
          version="1.1"
          viewBox="0 0 18 18"
          width="18px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              fillRule="nonzero"
              transform="translate(-938.000000, -206.000000)"
            >
              <path
                d="M945,206 C948.865993,206 952,209.134007 952,213 C952,214.662837 951.420201,216.190255 950.45174,217.391118 L955.03033,221.96967 C955.323223,222.262563 955.323223,222.737437 955.03033,223.03033 C954.764064,223.296597 954.3474,223.320803 954.053788,223.102948 L953.96967,223.03033 L949.391118,218.45174 C948.190255,219.420201 946.662837,220 945,220 C941.134007,220 938,216.865993 938,213 C938,209.134007 941.134007,206 945,206 Z M945,207.5 C941.962434,207.5 939.5,209.962434 939.5,213 C939.5,216.037566 941.962434,218.5 945,218.5 C948.037566,218.5 950.5,216.037566 950.5,213 C950.5,209.962434 948.037566,207.5 945,207.5 Z"
              />
            </g>
          </g>
        </svg>
      </button>
      <button
        className="styled__SearchButton-sc-12dq39v-3 eoLjh styled__PlainButton-sc-1119auh-0 kPdLbJ"
        data-experiment={true}
        value="Search"
      >
        <svg
          aria-hidden="true"
          height="18px"
          version="1.1"
          viewBox="0 0 18 18"
          width="18px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              fillRule="nonzero"
              transform="translate(-938.000000, -206.000000)"
            >
              <path
                d="M945,206 C948.865993,206 952,209.134007 952,213 C952,214.662837 951.420201,216.190255 950.45174,217.391118 L955.03033,221.96967 C955.323223,222.262563 955.323223,222.737437 955.03033,223.03033 C954.764064,223.296597 954.3474,223.320803 954.053788,223.102948 L953.96967,223.03033 L949.391118,218.45174 C948.190255,219.420201 946.662837,220 945,220 C941.134007,220 938,216.865993 938,213 C938,209.134007 941.134007,206 945,206 Z M945,207.5 C941.962434,207.5 939.5,209.962434 939.5,213 C939.5,216.037566 941.962434,218.5 945,218.5 C948.037566,218.5 950.5,216.037566 950.5,213 C950.5,209.962434 948.037566,207.5 945,207.5 Z"
              />
            </g>
          </g>
        </svg>
      </button>
    </form>
    <div
      className="styled__TextResizerDropdown-sc-12dq39v-16 gZvMfu Dropdown-rmc6yw-5 fBPFfC Dropdown__TabHiddenDropDown-rmc6yw-1 kbMrSv"
    >
      <button
        aria-label="Change text size"
        className="Dropdown__DropdownToggle-rmc6yw-0 gpiKKF Filters__Toggle-sc-1fn81fx-1 gsKueT Button__PlainButton-ayg7nk-2 chgswm"
        data-analytics-label="Change text size"
        onClick={[Function]}
      >
        <div
          tabIndex={-1}
        >
          <img
            alt=""
            aria-hidden="true"
            src="text-size.svg"
          />
        </div>
      </button>
      <div
        className="styled__TextResizerMenu-sc-12dq39v-17 eDJseE"
        tabIndex={0}
      >
        <label
          id="text-resizer-label"
        >
          Text size
        </label>
        <div
          className="controls"
        >
          <button
            aria-label="Decrease text size"
            className="styled__TextResizerChangeButton-sc-12dq39v-18 hPlFJJ styled__PlainButton-sc-1119auh-0 kPdLbJ"
            data-testid="decrease-text-size"
            onClick={[Function]}
          >
            <img
              alt=""
              aria-hidden="true"
              src="text-size-decrease.svg"
            />
          </button>
          <input
            aria-labelledby="text-resizer-label"
            data-testid="change-text-size"
            max={3}
            min={-2}
            onChange={[Function]}
            step="1"
            type="range"
            value={0}
          />
          <button
            aria-label="Increase text size"
            className="styled__TextResizerChangeButton-sc-12dq39v-18 hPlFJJ styled__PlainButton-sc-1119auh-0 kPdLbJ"
            data-testid="increase-text-size"
            onClick={[Function]}
          >
            <img
              alt=""
              aria-hidden="true"
              src="text-size-increase.svg"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="styled__MobileSearchWrapper-sc-12dq39v-11 foGdGm"
  >
    <hr
      className="styled__Hr-sc-12dq39v-12 iJYKyP"
    />
    <div
      className="styled__MobileSearchContainer-sc-12dq39v-10 hKhfgu"
    >
      <button
        className="styled__PlainButton-sc-1119auh-0 styled__SeachResultsTextButton-sc-12dq39v-15 styled__CloseSearchResultsTextButton-sc-12dq39v-19 kzdVbH"
        data-testid="close-search-results"
        onClick={[Function]}
      >
        <div
          className="styled__InnerText-sc-12dq39v-14 fahaRL"
        >
          Close
        </div>
      </button>
      <form
        action="#"
        className="styled__SearchInputWrapper-sc-12dq39v-7 bHZhxZ"
        data-experiment={true}
        data-testid="mobile-search"
        onSubmit={[Function]}
      >
        <input
          aria-label="Search this book"
          autoFocus={true}
          className="styled__SearchInput-sc-12dq39v-8 bHFdA"
          data-testid="mobile-search-input"
          onChange={[Function]}
          placeholder="Search this book"
          type="search"
          value=""
        />
        <button
          className="styled__SearchButton-sc-12dq39v-3 eoLjh styled__PlainButton-sc-1119auh-0 kPdLbJ"
          data-experiment={true}
          value="Search"
        >
          <svg
            aria-hidden="true"
            height="18px"
            version="1.1"
            viewBox="0 0 18 18"
            width="18px"
          >
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <g
                fill="currentColor"
                fillRule="nonzero"
                transform="translate(-938.000000, -206.000000)"
              >
                <path
                  d="M945,206 C948.865993,206 952,209.134007 952,213 C952,214.662837 951.420201,216.190255 950.45174,217.391118 L955.03033,221.96967 C955.323223,222.262563 955.323223,222.737437 955.03033,223.03033 C954.764064,223.296597 954.3474,223.320803 954.053788,223.102948 L953.96967,223.03033 L949.391118,218.45174 C948.190255,219.420201 946.662837,220 945,220 C941.134007,220 938,216.865993 938,213 C938,209.134007 941.134007,206 945,206 Z M945,207.5 C941.962434,207.5 939.5,209.962434 939.5,213 C939.5,216.037566 941.962434,218.5 945,218.5 C948.037566,218.5 950.5,216.037566 950.5,213 C950.5,209.962434 948.037566,207.5 945,207.5 Z"
                />
              </g>
            </g>
          </svg>
        </button>
        
        
      </form>
      <div
        className="styled__TextResizerDropdown-sc-12dq39v-16 gZvMfu Dropdown-rmc6yw-5 fBPFfC Dropdown__TabHiddenDropDown-rmc6yw-1 kbMrSv"
      >
        <button
          aria-label="Change text size"
          className="Dropdown__DropdownToggle-rmc6yw-0 gpiKKF Filters__Toggle-sc-1fn81fx-1 liapzj Button__PlainButton-ayg7nk-2 chgswm"
          data-analytics-label="Change text size"
          onClick={[Function]}
        >
          <div
            tabIndex={-1}
          >
            <img
              alt=""
              aria-hidden="true"
              src="text-size.svg"
            />
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
